### PR TITLE
Post new task

### PR DESCRIPTION
**Pull request by junydania**
## What was done:
    - Feature to post a new task without signing in
   - At the point of creating the task, user must be authenticated
 
## What was learnt:
   - Overriding devise custom after_sign_in_path_for(resource) method

## Pivotal Tracker
https://www.pivotaltracker.com/story/show/146678435

### CODE REVIEW

File: app/views/tasks/new.html.haml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  .field
+    = f.label :minimum_budget, "Minimum Budget($):"
+    %br/
+    = f.number_field :minimum_budget
+  .field
+    = f.label :maximum_budget, "Maximum Budget($):"
+    %br/
+    = f.number_field :maximum_budget
+  .field
+    = f.label :location, "Location"
```
Comment (ceca2017) [inline app/views/tasks/new.html.haml]: Perhaps move this up next to description (as it is required)
Perhaps add a star to the label to indicate that it is required (and add an explanation at the bottom)

File: spec/models/task_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   it { is_expected.to validate_presence_of :location }
+ end
+
+
+  describe 'Factory' do
+    it 'should have valid Factory' do
+      expect(FactoryGirl.create(:task)).to be_valid
+    end
+  end
+end
```
Comment (ceca2017) [inline spec/models/task_spec.rb:29]: Add tests to check assoications

File: spec/models/task_category_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  describe 'DB table task_categories' do
+    it { is_expected.to have_db_column :name }
+  end
+
+  describe 'Factory' do
+    it 'should have valid Factory' do
+      expect(FactoryGirl.create(:task_category)).to be_valid
+    end
+  end
+end
```
Comment (ceca2017) [inline spec/models/task_category_spec.rb:17]: Add check for associations

**Review by ceca2017 — COMMENTED** [review]
Please check comments

File: spec/models/task_category_spec.rb (hunk 3 above)
Comment (junydania) [reply spec/models/task_category_spec.rb:17]: Done

File: spec/models/task_spec.rb (hunk 2 above)
Comment (junydania) [reply spec/models/task_spec.rb:29]: Done

File: app/views/tasks/new.html.haml (hunk 1 above)
Comment (junydania) [reply app/views/tasks/new.html.haml]: Field moved and added a "required field attribute" to the field to let the user know it is a mandatory field.

**Review by ceca2017 — APPROVED** [review]
Great stuff!

**Pull request merged**